#!/usr/bin/env python
PACKAGE = "rc_pick_client"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# name, type,level, description, default, min, max
gen.add("max_grasps",      int_t, 0, "Maximum number of provided grasps", 5, 1, 20)

gen.add("cluster_max_dimension", double_t, 0, "Diameter of the largest sphere enclosing each cluster in meters. Clusters larger than this value are filtered out before grasp computation.", 0.3, 0.05, 0.8)
gen.add("cluster_max_curvature",      double_t, 0, "Maximum curvature allowed within one cluster. The smaller this value, the more clusters will be split apart.", 0.11, 0.005, 0.5)
gen.add("clustering_patch_size",      int_t, 0, "Size in pixels of the square patches the depth map is subdivided into during the first clustering step", 4,3,10)
gen.add("clustering_max_surface_rmse",      double_t, 0, "Maximum root-mean-square error (RMSE) in meters of points belonging to a surface", 0.004,0.0005,0.01)
gen.add("clustering_discontinuity_factor",      double_t, 0, "Factor used to discriminate depth discontinuities within a patch. The smaller this value, the more clusters will be split apart.", 1,0.5,5)

exit(gen.generate(PACKAGE, "rc_pick_client", "pickModule"))
